akka-remote/src/main/scala/akka/remote/serialization/MiscMessageSerializer.scala

```diff
@@ -0,0 +1,110 @@
+  }
+
+  private def serializeIdentify(identify: Identify): Array[Byte] =
+    ContainerFormats.Identify.newBuilder()
+      .setMessageId(payloadBuilder(identify.messageId))
+      .build()
+      .toByteArray
+
+  private def serializeActorIdentity(actorIdentity: ActorIdentity): Array[Byte] = {
+    val builder =
+      ContainerFormats.ActorIdentity.newBuilder()
+        .setCorrelationId(payloadBuilder(actorIdentity.correlationId))
+
+    actorIdentity.ref.foreach { actorRef ⇒
+      builder.setRef(actorRefBuilder(actorRef))
+    }
+
+    builder
+      .build()
+      .toByteArray
+  }
+
+  private def actorRefBuilder(actorRef: ActorRef): ContainerFormats.ActorRef.Builder =
+    ContainerFormats.ActorRef.newBuilder()
+      .setPath(Serialization.serializedActorPath(actorRef))
+
+  private def payloadBuilder(input: Any): ContainerFormats.Payload.Builder = {
+    val payload = input.asInstanceOf[AnyRef]
+    val builder = ContainerFormats.Payload.newBuilder()
+    val serializer = serialization.findSerializerFor(payload)
+
+    builder
+      .setEnclosedMessage(ByteString.copyFrom(serializer.toBinary(payload)))
+      .setSerializerId(serializer.identifier)
+
+    serializer match {
+      case ser2: SerializerWithStringManifest ⇒
+        val manifest = ser2.manifest(payload)
+        if (manifest != "")
+          builder.setMessageManifest(ByteString.copyFromUtf8(manifest))
+      case _ ⇒
+        if (serializer.includeManifest)
+          builder.setMessageManifest(ByteString.copyFromUtf8(payload.getClass.getName))
+    }
+
+    builder
+  }
+
+  override def manifest(o: AnyRef): String =
+    o.getClass.getName
+
+  private val IdentifyClassName = classOf[Identify].getName
+  private val ActorIdentityClassName = classOf[ActorIdentity].getName
+
+  override def fromBinary(bytes: Array[Byte], manifest: String): AnyRef =
+    manifest match {
+      case IdentifyClassName      ⇒ deserializeIdentify(bytes)
+      case ActorIdentityClassName ⇒ deserializeActorIdentity(bytes)
+      case unsupported            ⇒ throw new IllegalArgumentException(s"Cannot deserialize object of type [${unsupported}]")
+    }
+
+  private def deserializeIdentify(bytes: Array[Byte]): Identify = {
+    val identifyProto = ContainerFormats.Identify.parseFrom(bytes)
+    val messageId = deserializePayload(identifyProto.getMessageId)
+    Identify(messageId)
+  }
+
+  private def deserializeActorIdentity(bytes: Array[Byte]): ActorIdentity = {
+    val actorIdentityProto = ContainerFormats.ActorIdentity.parseFrom(bytes)
+    val correlationId = deserializePayload(actorIdentityProto.getCorrelationId)
+    val actorRef =
+      if (actorIdentityProto.getRef != null &&
+        actorIdentityProto.getRef.getPath != null &&
+        actorIdentityProto.getRef.getPath != "")
+        Some(deserializeActorRef(actorIdentityProto.getRef))
+      else
+        None
```

Comment:
@patriknw - using if/else as suggested with some difference due to `actorIdentityProto` is a protobuf object, and the only way to check if `ref` is defined is by using the null checks above